test/unit/lodash.spec.js

```diff
@@ -0,0 +1,275 @@
+            expect(array.length).to.eql(5);
+            expect(_.filter.apply(_, datasets.filter2())).to.eql([{name: 'Bow', age: 23, active: true},
+                {name: 'Row', age: 23, active: true},
+                {name: 'Sou', age: 23, active: true}]);
+        });
+
+    });
+
+    describe('find', function () {
+        it('should find Helen', function () {
+            var testArray = _.fill(new Array(50), 1).map(function () {
+                if (Math.random() < 0.2) {
+                    return {name: 'Kodi', age: 12, active: true};
+                } else if (Math.random() < 0.4) {
+                    return {name: 'Olaf', age: 32, active: false};
+                } else if (Math.random() < 0.6) {
+                    return {name: 'Helen', age: 65, active: true};
+                } else if (Math.random() < 0.8) {
+                    return {name: 'Pluto', age: 25, active: false};
+                } else {
+                    return {name: 'Rudi', age: 43, active: true};
+                }
+            });
+            var params = datasets.find(testArray);
+            var array = params[0];
+            var method = params[1];
+            var result = _.find.apply(_, params);
+            expect(array.length).to.eql(50);
+            expect(Object.keys(method).length).to.eql(1);
+            expect(_.values(result)).to.contain('Helen');
+            expect(_.values(result)).to.contain(65);
+        });
+        it('should return first matched element for name property', function () {
+            var params = datasets.find2();
+            var array = params[0];
+            expect(array.length).to.eql(3);
+            array.forEach(function (n) {
+                expect(_.has(n, 'NAME')).to.eql(true);
+                expect(_.has(n, 'active')).to.eql(true);
+            });
+            expect(_.find.apply(_, datasets.find2())).to.eql({
+                NAME: 'Rudi',
+                age: 15,
+                active: true,
+                gender: 'M'
+            });
+        });
+    });
+
+    describe('groupBy', function () {
+        it('should group floating-point numbers with Math function', function () {
+            var params = datasets.groupBy();
+            var array = params[0];
+            expect(array.length).to.eql(7);
+            expect(_.groupBy.apply(_, datasets.groupBy())).to.eql({
+                3: [3.3],
+                4: [4.2, 4.6],
+                6: [6.1, 6.3],
+                8: [8.3, 8.9]
+            });
+        });
+        it('should group all given things by length', function () {
+            var params = datasets.groupBy2();
+            var array = params[0];
+            expect(array.length).to.eql(6);
+            expect(_.groupBy.apply(_, datasets.groupBy2())).to.eql({
+                3: ['xyz', 'xyz'],
+                5: ['combi'],
+                undefined: [2, 4, 5]
+            });
+        });
+    });
+
+    describe('includes', function () {
```

Comment:
includes: Properties in tests have no semantic meaning because I can just put some random values, however in terms of scenario it is ok(but scenario is poor to be honest). I expected second one to be more advanced but it not only didn't test scenario properly and missed many important parts. You have completly forgot to test fromIndex paramter and other parameter types like in most of the exercises.